internal/k8s/controllers/config_controller_test.go

```diff
@@ -0,0 +1,335 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	v1beta1 "go.universe.tf/metallb/api/v1beta1"
+	v1beta2 "go.universe.tf/metallb/api/v1beta2"
+	"go.universe.tf/metallb/internal/config"
+	metallbcfg "go.universe.tf/metallb/internal/config"
+	"go.universe.tf/metallb/internal/pointer"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestConfigController(t *testing.T) {
+	var (
+		testNamespace = "test-controller"
+		scheme        = runtime.NewScheme()
+		// Defining config using all features.
+		resources = metallbcfg.ClusterResources{
```

Comment:
I don't think we need to have this so fully detailed.
What we want is to make sure that all the crds are passed to config.For , so it's sufficient to have the names.
If we need to have the full details because config.For would fail otherwise, let's move this as a global to the bottom of the file, so it won't distract from the test (also, give it a name so it's clear that we don't care about the content, like "validConfig". Also also, I'd test the scenario where config.For returns error, so it makes sense to have also an "invalidConfig". At that point, config can be a parameter of the test

Reply:
done